libs/rtefsutils/src/RteFsUtils.cpp

```diff
@@ -269,15 +270,41 @@ bool RteFsUtils::SetFileReadOnly(const string& path, bool bReadOnly) {
  */
 
   error_code ec;
-  fs::perms perms;
+  constexpr fs::perms write_mask = fs::perms::owner_write | fs::perms::group_write | fs::perms::others_write;
 
-  // Set file permissions
   if (bReadOnly) {
-    perms = fs::perms::owner_read | fs::perms::group_read | fs::perms::others_read;
+    // Remove write bits
+    fs::permissions(fs::path(path), write_mask, fs::perm_options::remove | fs::perm_options::nofollow, ec);
   } else {
-    perms = fs::perms::all;
+    // Inherit write bits from parent directory
+    fs::perms perms = fs::status(fs::path(path).parent_path(), ec).permissions() & write_mask;
+
+    // If parent directory is not writable, lets rely on current umask
+    if (perms == fs::perms::none) {
+      // Only way to get current umask is to change it, so revert the possible change directly.
+      mode_t umask = ::umask(0);
```

Comment:
It does not appear to be a way to retrieve the umask value without actually changing it.
Is it fine to do it like this or should we find a different way to get the value?

Reply:
I raised this question with some C++ experts here. It looks weird but if it works we might keep it for the time being.